src/main/java/edu/cornell/kfs/tax/dataaccess/impl/TransactionRowBuilder.java

```diff
@@ -334,7 +337,7 @@ Document getWorkflowDocumentForTaxRow(String documentId, T summary) {
                     // If still greater than cached ID and more unfetched docs exist, perform next bulk retrieval.
                     if (idCompareResult > 0 && documentIdsForBulkQuery.hasNext()) {
                         StringBuilder docIdCriteria = new StringBuilder(DOC_ID_CRITERIA_SIZE);
-                        for (int i = 0; documentIdsForBulkQuery.hasNext() && i < DOC_SEARCH_BATCH_SIZE; i++) {
+                        for (int i = 0; documentIdsForBulkQuery.hasNext() && i < getMaxSearchSize(); i++) {

```

Comment:
You may want to store the result of getMaxSearchSize() in a local variable, otherwise the method will get called on every iteration of the loop. Unless you think the parameter retrieval caching in that method can mitigate efficiency concerns (though I'm not certain if it can cache null values for nonexistent parameters).